src/backend/tools/go.CLI/cmd/gocaliopen/cli_cmds/root.go

```diff
@@ -0,0 +1,272 @@
+ * // Use of this source code is governed by a GNU AFFERO GENERAL PUBLIC
+ * // license (AGPL) that can be found in the LICENSE file.
+ */
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	. "github.com/CaliOpen/Caliopen/src/backend/defs/go-objects"
+	"github.com/CaliOpen/Caliopen/src/backend/interfaces/REST/go.server"
+	"github.com/CaliOpen/Caliopen/src/backend/main/go.backends/cache/redis"
+	"github.com/CaliOpen/Caliopen/src/backend/main/go.backends/index/elasticsearch"
+	"github.com/CaliOpen/Caliopen/src/backend/main/go.backends/store/cassandra"
+	"github.com/CaliOpen/Caliopen/src/backend/main/go.main/facilities/REST"
+	"github.com/CaliOpen/Caliopen/src/backend/protocols/go.smtp"
+	log "github.com/Sirupsen/logrus"
+	"github.com/gocql/gocql"
+	"github.com/nats-io/go-nats"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"net/http"
+	"os"
+)
+
+type CmdConfig struct {
+	rest_api.APIConfig
+	rest_api.IndexConfig
+	rest_api.ProxyConfig
+}
+
+var (
+	cfgPath     string
+	apiCfgFile  string
+	lmtpCfgFile string
+	apiConf     CmdConfig
+	lmtpConf    caliopen_smtp.SMTPConfig
+
+	// RootCmd represents the base command when called without any subcommands
+	RootCmd = &cobra.Command{
+		Use:   "gocaliopen",
+		Short: "Caliopen CLI to interact with stack",
+		Long: `gocaliopen needs two of Caliopen's config files : caliopen-go-api_dev.yaml and caliopen-go-lmtp_dev.yaml.
+It loads them from within directory specified with flag --confPath, or if path/filenames are specified with the --apiConf and --lmtpConf.
+gocaliopen subcommands could interact with
+	- store (Cassandra)
+	- index (Elasticsearch)
+	- message queue (NATS)
+	- cache (Redis)
+	- caliopen REST facility	
+	- apiV1
+	- apiV2
+	- lmtpd
+	- caliopen notification facility
+	- email broker	`,
+	}
+)
+
+const __version__ = "0.10.0"
+
+func init() {
+	cobra.OnInitialize(initConfig)
+
+	RootCmd.PersistentFlags().StringVar(&cfgPath, "confPath", "", "Path to seek the two mandatory config files: caliopen-go-api_dev.yaml and caliopen-go-lmtp_dev.yaml")
+	RootCmd.PersistentFlags().StringVar(&apiCfgFile, "apiConf", "", "Caliopen's API config file (default is ./caliopen-go-api_dev.yaml)")
+	RootCmd.PersistentFlags().StringVar(&lmtpCfgFile, "lmtpConf", "", "Caliopen's lmtpd config file (default is ./caliopen-go-lmtp_dev.yaml)")
+}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	if err := RootCmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+// initConfig reads in config files and ENV variables if set.
+func initConfig() {
+	apiCfg := viper.New()
+	lmtpCfg := viper.New()
+
+	if cfgPath != "" {
+		// Use path from the flag
+		apiCfg.AddConfigPath(cfgPath)
+		lmtpCfg.AddConfigPath(cfgPath)
+	} else {
+		// set defaults to relative path from /tools/go.CLI/cmd/gocaliopen
+		apiCfg.AddConfigPath("../../../../configs")
```

Comment:
I still really don't like this relative path usage that are really uncommon. Can we remove such default values. Yes we will have to provide configPath or have a correct environment, that's is better than this usage.